datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -376,7 +381,8 @@ impl ExecutionPlan for HashJoinExec {
     ) -> Result<SendableRecordBatchStream> {
         let on_left = self.on.iter().map(|on| on.0.clone()).collect::<Vec<_>>();
         let on_right = self.on.iter().map(|on| on.1.clone()).collect::<Vec<_>>();
-
+        let hashjoin_metrics = HashJoinMetrics::new(partition, &self.metrics);
+        let timer = hashjoin_metrics.build_time.timer();
         let left_fut = match self.mode {
```

Comment:
Is it correct to calculate the build time here? The `left_fut` returned here is a `Future`.

Reply:
> Is it correct to calculate the build time here? The `left_fut` returned here is a `Future`.

Thanks！I'll check this code later.